gen.py

```diff
@@ -67,10 +73,45 @@ def generate_single_tag(args: Any):
     repo = Repo.default()
     repo.fetch()
     version = args.version
-    _generate_tag(repo, version)
+    _generate_tag(repo, version, force=args.force)
     _generate_list_of_packages()
 
 
+def _nix_prefetch_git(url: str, rev: str) -> str:
+    return _run(f"nix-prefetch-git {url} --rev {rev} --quiet").decode("utf-8")
+
+
+def _prefetch_output_hashes(cargo_lock: str) -> str:
+    output_hashes = set()
+    for package in tomllib.loads(cargo_lock)["package"]:
+        source = package.get("source", "")
+        if not source.startswith("git+"):
```

Comment:
Let's merge this check with the `git_url_re.match` check

Reply:
I didn't to begin with because it's less performant to do the match vs. a startswith and we expect relatively few git dependencies.

Still, they're shortish strings and the cargo.lock files seem to only have ~400 of them, so it shouldn't be too big of a difference. I'll add a comment suggest this alternative in case it is hurting perf in the future,